File: examples/javascript/raw_send.js
const { PocketOption } = require("./binary-options-tools.node");

async function main(ssid) {
  // Initialize the API client
  const api = new PocketOption(ssid);

  // Wait for connection to establish
  await new Promise((resolve) => setTimeout(resolve, 5000));

  try {
    // Subscribe to signals
    await api.sendRawMessage('42["signals/subscribe"]');
    console.log("Sent signals subscription message");

    // Subscribe to price updates
    await api.sendRawMessage('42["price/subscribe"]');
    console.log("Sent price subscription message");

    // Custom message example
    const customMessage = '42["custom/event",{"param":"value"}]';
    await api.sendRawMessage(customMessage);
    console.log(`Sent custom message: ${customMessage}`);

    // Multiple messages in sequence
    const messages = [
      '42["chart/subscribe",{"asset":"EURUSD"}]',
      '42["trades/subscribe"]',
      '42["notifications/subscribe"]',
    ];

    for (const msg of messages) {
      await api.sendRawMessage(msg);
      console.log(`Sent message: ${msg}`);
      await new Promise((resolve) => setTimeout(resolve, 1000)); // 1 second delay
    }
  } catch (error) {
    console.log(`Error sending message: ${error}`);
  }
}

// Check if ssid is provided as command line argument
const ssid = "";

main(ssid).catch(console.error);

File: examples/python/async/raw_send.py
from BinaryOptionsToolsV2.pocketoption import PocketOptionAsync
import asyncio


async def main(ssid: str):
    # Initialize the API client
    api = PocketOptionAsync(ssid)
    await asyncio.sleep(5)  # Wait for connection to establish

    # Example of sending a raw message
    try:
        # Subscribe to signals
        await api.send_raw_message('42["signals/subscribe"]')
        print("Sent signals subscription message")

        # Subscribe to price updates
        await api.send_raw_message('42["price/subscribe"]')
        print("Sent price subscription message")

        # Custom message example
        custom_message = '42["custom/event",{"param":"value"}]'
        await api.send_raw_message(custom_message)
        print(f"Sent custom message: {custom_message}")

        # Multiple messages in sequence
        messages = [
            '42["chart/subscribe",{"asset":"EURUSD"}]',
            '42["trades/subscribe"]',
            '42["notifications/subscribe"]',
        ]

        for msg in messages:
            await api.send_raw_message(msg)
            print(f"Sent message: {msg}")
            await asyncio.sleep(1)  # Small delay between messages

    except Exception as e:
        print(f"Error sending message: {e}")


if __name__ == "__main__":
    ssid = input("Please enter your ssid: ")
